packages/pluggableWidgets/datagrid-web/src/utils/settings.ts

```diff
@@ -27,15 +27,18 @@ export function createSettings(
     { columnOrder, hiddenColumns, sortBy, filters, widths }: Settings,
     columns: Array<{ header: string; id: string }>
 ): PersistedSettings[] {
-    return columns.map(column => ({
-        column: column.header,
-        sort: !!sortBy.find(s => s.id === column.id),
-        sortMethod: sortBy.find(s => s.id === column.id)?.desc ? "desc" : "asc",
-        filter: filters.find(f => f.id === column.id)?.value ?? "",
-        hidden: !!hiddenColumns.find(h => h === column.id),
-        order: columnOrder.findIndex(o => o === column.id),
-        width: widths[column.id]
-    }));
+    return columns.map((column, index) => {
+        const columnIndex = columnOrder.findIndex(o => o === column.id);
+        return {
+            column: column.header,
+            sort: !!sortBy.find(s => s.id === column.id),
+            sortMethod: sortBy.find(s => s.id === column.id)?.desc ? "desc" : "asc",
+            filter: filters.find(f => f.id === column.id)?.value ?? "",
+            hidden: !!hiddenColumns.find(h => h === column.id),
+            order: columnIndex > -1 ? columnIndex : index,
```

Comment:
Why can column index be -1? Is is possible for `columnOrder` to not contain all columns?

Reply:
Because initially the react-table returns empty array if nothing has been change related to column order

Reply:
And it doesnt call again setValue from settings because the settings are not changed, as check in line 117 of the hook